### PR TITLE
[flutter_tools] Launch DevTools with 'v'

### PR DESCRIPTION
**Pull request by zanderso**
## Description

Launch DevTools from the resident runner with 'v' when there is a vm service available.

## Related Issues

https://github.com/flutter/flutter/issues/41407

## Tests

I added the following tests:

I updated tests of the command help.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -982,6 +985,33 @@ abstract class ResidentRunner {
     }
   }
 
+  io.HttpServer _devtoolsServer;
+
+  Future<void> launchDevTools() async {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/resident_runner.dart:990]: For testing: what I would do is give the ResidentRunner an optional constructor argument:

```
Future<io.HttpServer> Function({bool enableStdinCommands}) serveDevTools
```

Then in the launchDevTools method do something like:

```
await (serveDevTools ?? devtools_server.serveDevTools)(enabledStinCommands: false)
```

That way we can test it without adding more to the context in the short term

Comment (zanderso) [reply packages/flutter_tools/lib/src/resident_runner.dart:990]: After further thought, not much of the behavior of the devtools server and its client in the browser can be tested. Instead, I added a test to `terminal_handler_test.dart` that ensures that `ResidentRunner.launchDevTools` is called when we get 'v' on the terminal.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart:990]: Yeah seems reasonable to me

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by jacob314** [comment]
This is great.
It would be nice to display a link to this URL in the console rather than 
"An Observatory debugger and profiler on iPhone 11 Pro Max is available at:
http://127.0.0.1:56450/asdfasdfw=/"
We could provide a link from DevTools to the observatory as an affordance for advanced users who need functionality not yet in DevTools.

**Pull request merged**